File: CS/TreeListDragDropMultipleNodes/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using TreeListDragDropMultipleNodes.Models;

namespace TreeListDragDropMultipleNodes.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        [ValidateInput(false)]
        public ActionResult TreeListPartial()
        {
            return PartialView("_TreeListPartial", DataHelper.GetData());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult TreeListDragDropAction(int id, object[] nodeKeys, int parentID)
        {
            if (nodeKeys == null)
                DataHelper.MoveNode(id, parentID);
            else
                DataHelper.MoveNodes(nodeKeys, parentID);

            return PartialView("_TreeListPartial", DataHelper.GetData());
        }
    }
}

File: VB/TreeListDradDropSelectedNodes/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        Return View()
    End Function

    <ValidateInput(False)>
    Public Function TreeListPartial() As ActionResult
        Return PartialView("_TreeListPartial", DataHelper.GetData())
    End Function

    <HttpPost, ValidateInput(False)>
    Public Function TreeListDragDropAction(ByVal id As Integer, ByVal nodeKeys As Integer(), ByVal parentID As Integer) As ActionResult
        If nodeKeys Is Nothing Then
            DataHelper.MoveNode(id, parentID)
        Else
            DataHelper.MoveNodes(nodeKeys, parentID)
        End If

        Return PartialView("_TreeListPartial", DataHelper.GetData())
    End Function
End Class